src/field_theory/adjoin.lean

```diff
@@ -61,58 +120,55 @@ instance adjoin.field_coe : has_coe_t F (adjoin F S) :=
 {coe := λ x, ⟨algebra_map F E x, adjoin.algebra_map_mem F S x⟩}
 
 lemma subset_adjoin : S ⊆ adjoin F S :=
-λ x hx, field.mem_closure (or.inr hx)
+λ x hx, subfield.subset_closure (or.inr hx)
 
 instance adjoin.set_coe : has_coe_t S (adjoin F S) :=
 {coe := λ x, ⟨x,subset_adjoin F S (subtype.mem x)⟩}
 
-lemma adjoin.mono (T : set E) (h : S ⊆ T) : (adjoin F S : set E) ⊆ adjoin F T :=
-field.closure_mono (set.union_subset (set.subset_union_left _ _)
+lemma adjoin.mono (T : set E) (h : S ⊆ T) : adjoin F S ≤ adjoin F T :=
+subfield.closure_mono (set.union_subset (set.subset_union_left _ _)
   (set.subset_union_of_subset_right h _))
 
-instance adjoin.is_subfield : is_subfield (adjoin F S : set E) := field.closure.is_subfield
-
---Lean has trouble figuring this out on its own
-instance adjoin.is_field : field (adjoin F S) := @is_subfield.field E _ ((adjoin F S) : set E) _
-
-lemma adjoin_contains_field_as_subfield (F : set E) {HF : is_subfield F} : F ⊆ adjoin F S :=
+lemma adjoin_contains_field_as_subfield (F : subfield E) : (F : set E) ⊆ adjoin F S :=
 λ x hx, adjoin.algebra_map_mem F S ⟨x, hx⟩
 
+lemma subset_adjoin_of_subset_left {F : subfield E} {T : set E} (HT : T ⊆ F) :
+  T ⊆ adjoin F S :=
+λ x hx, (adjoin F S).algebra_map_mem ⟨x, HT hx⟩
+
 lemma subset_adjoin_of_subset_right {T : set E} (H : T ⊆ S) : T ⊆ adjoin F S :=
 begin
   intros x hx,
   exact subset_adjoin F S (H hx),
 end
 
-/-- If `K` is a field with `F ⊆ K` and `S ⊆ K` then `adjoin F S ⊆ K`. -/
-lemma adjoin_subset_subfield {K : set E} [is_subfield K] (HF : set.range (algebra_map F E) ⊆ K)
-  (HS : S ⊆ K) : (adjoin F S : set E) ⊆ K :=
+/-- If `K` is a field with `F ⊆ K` and `S ⊆ K` then `adjoin F S ≤ K`. -/
+lemma adjoin_le_subfield {K : subfield E} (HF : set.range (algebra_map F E) ⊆ K)
+  (HS : S ⊆ K) : (adjoin F S).to_subfield ≤ K :=
 begin
-  apply field.closure_subset,
+  apply subfield.closure_le.mpr,
   rw set.union_subset_iff,
   exact ⟨HF, HS⟩,
 end
 
 /-- `S ⊆ adjoin F T` if and only if `adjoin F S ⊆ adjoin F T`. -/
-lemma adjoin_subset_iff {T : set E} : S ⊆ adjoin F T ↔ (adjoin F S : set E) ⊆ adjoin F T :=
-⟨λ h, adjoin_subset_subfield F S (adjoin.range_algebra_map_subset F T) h,
-  λ h, set.subset.trans (subset_adjoin F S) h⟩
-
-lemma subfield_subset_adjoin_self {F : set E} {HF : is_subfield F} {T : set E} {HT : T ⊆ F} :
-  T ⊆ adjoin F S :=
-λ x hx, adjoin.algebra_map_mem F S ⟨x,HT hx⟩
+lemma subset_adjoin_iff {T : set E} : S ≤ adjoin F T ↔ adjoin F S ≤ adjoin F T :=
```

Comment:
Can't the right hand side be any intermediate field? It doesn't have to be an `adjoin F T`, right?
If so, please swap the sides of the `iff` and make this a simp lemma.

Reply:
Actually, the generalized version (with any intermediate field) is already present (`adjoin_le_iff` at the top of the file) so I'll just delete `subset_adjoin_iff` and make `adjoin_le_iff` into a simp lemma